rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMModelImpl.java

```diff
@@ -119,10 +122,43 @@
         return this.nodes;
     }
 
-    void setNodes(HashMap<String, NodeSource> nodes) {
-        this.nodes = nodes;
+    void setNodes(Map<String, NodeSource> nodes) {
```

Comment:
Why do we need a deep copy? I don 't understand the scope of this changes. Also we should not have TODO comments in the code, we shouldn't postpone the right implementations for later.